src/main/java/org/apelsin/tutorial/collections/ArrayCollection.java

```diff
@@ -46,4 +48,39 @@ public void doubleContentLengthIfCapacityIsReached() {
         }
     }
 
+    public Object remove(int index) throws InstanceNotFoundException {
+        if (index < 0 || index >= content.length)
+            throw new InstanceNotFoundException();
+        Object oldValue = content[index];
+        Object[] temp = new Object[content.length - 1];
+
+        for (int i = 0, j = 0; i < content.length; i++) {
+            if (i != index)
+                temp[j++] = content[i];
+        }
+
+        content = temp;
+        return oldValue;
+    }
+
+    public Object remove(Object value){
+        boolean found = false;
+        int indexOfValue = 0;
+        for (int i = 0; i < content.length; i++) {
+            if (content[i].equals(value)){
+                found = true;
```

Comment:
Bad practice use length of array in the ArrayList because length and size may be different